### PR TITLE
Core/Quests: Some quest related cp (Part1)

### PR DESCRIPTION
**Pull request by Faq**
This PR starts a quest line of some OPCODE convert journey and is actually a requirement for PR (Part2).

**Changes proposed:**

**This PR (Part1):**
- cherry-pick 770ff57101e1ca6cce0922c3835de527f47d2c9e
- cherry-pick part of 10c9c557003dd6862295c2e9c792d72d4b7cd9c3

Work continues:
- Next PR (Part2): SMSG_QUESTGIVER_OFFER_REWARD
- Next PR (Part3): SMSG_QUESTGIVER_QUEST_DETAILS

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)
Compiles
Tested with new dk character and quest chain its starting zone, where [q chain](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?quest=12678) rewards are like: rune forge skill, key, objective count on spell use, mob slain etc.

**Known issues and TODO list:** (add/remove lines as needed)

- [x] none

### CODE REVIEW

**Comment by jackpoz** [comment]
> **Tests performed:** (Does it build, tested in-game, etc.)
> Compiles

every affected opcode should be tested ingame before this kind of PRs can be merged, otherwise we move from a working status to a bugged status

**Comment by Faq** [comment]
Added test info in description.

File: src/server/game/Entities/Creature/GossipDef.cpp [hunk 1]
```diff
@@ -432,71 +432,10 @@ void PlayerMenu::SendQuestGiverQuestDetails(Quest const* quest, ObjectGuid npcGU
     data << uint32(quest->GetSuggestedPlayers());
     data << uint8(0);                                       // CGQuestInfo::m_startQuestCheat
 
-    if (quest->HasFlag(QUEST_FLAGS_HIDDEN_REWARDS))
```
Comment (Faq) [inline src/server/game/Entities/Creature/GossipDef.cpp:435]: https://github.com/TrinityCore/WowPacketParser/blob/6fc34f1deb4d92b385886495ca98c21cc772e71c/WowPacketParser/Parsing/Parsers/QuestHandler.cs#L861

File: src/server/game/Quests/QuestDef.cpp [hunk 2]
```diff
@@ -263,6 +263,57 @@ bool Quest::IsAutoAccept() const
...
+    for (uint8 i = 0; i < QUEST_REWARDS_COUNT; ++i)
+    {
+        if (ItemTemplate const* itemTemplate = sObjectMgr->GetItemTemplate(RewardItemId[i]))
+            data << uint32(itemTemplate->DisplayInfoID);
+        else
+            data << uint32(0);
+    }
+
+    data << uint32(GetRewOrReqMoney());
+    data << uint32(GetXPReward(player) * sWorld->getRate(RATE_XP_QUEST));
```
Comment (Faq) [inline src/server/game/Quests/QuestDef.cpp:295]:
```suggestion
    if (HasFlag(QUEST_FLAGS_HIDDEN_REWARDS))
    {
        data << uint32(0);                                  // Rewarded chosen items hidden
        data << uint32(0);                                  // Rewarded items hidden
        data << uint32(0);                                  // Rewarded money hidden
        data << uint32(0);                                  // Rewarded XP hidden
    }
    else
    {
        data << uint32(GetRewChoiceItemsCount());
        for (uint8 i = 0; i < QUEST_REWARD_CHOICES_COUNT; ++i)
            data << uint32(RewardChoiceItemId[i]);
        for (uint8 i = 0; i < QUEST_REWARD_CHOICES_COUNT; ++i)
            data << uint32(RewardChoiceItemCount[i]);
        for (uint8 i = 0; i < QUEST_REWARD_CHOICES_COUNT; ++i)
        {
            if (ItemTemplate const* itemTemplate = sObjectMgr->GetItemTemplate(RewardChoiceItemId[i]))
                data << uint32(itemTemplate->DisplayInfoID);
            else
                data << uint32(0);
        }

        data << uint32(GetReqItemsCount());
        for (uint8 i = 0; i < QUEST_REWARDS_COUNT; ++i)
            data << uint32(RewardItemId[i]);
        for (uint8 i = 0; i < QUEST_REWARDS_COUNT; ++i)
            data << uint32(RewardItemIdCount[i]);
        for (uint8 i = 0; i < QUEST_REWARDS_COUNT; ++i)
        {
            if (ItemTemplate const* itemTemplate = sObjectMgr->GetItemTemplate(RewardItemId[i]))
                data << uint32(itemTemplate->DisplayInfoID);
            else
                data << uint32(0);
        }

        data << uint32(GetRewOrReqMoney());
        data << uint32(GetXPReward(player) * sWorld->getRate(RATE_XP_QUEST));
    }
```

**Comment by Aokromes** [comment]
https://github.com/TrinityCore/TrinityCore/pull/25232

**Pull request closed without merging**